lighthouse-core/test/report/html/renderer/category-renderer-test.js

```diff
@@ -112,6 +112,38 @@ describe('CategoryRenderer', () => {
     assert.ok(warningEl.textContent.includes(auditResult.warnings[1]), '2nd warning provided');
   });
 
+  it('expands passed audit group when warnings exist', () => {
+    const category = sampleResults.reportCategories.find(c => c.id === 'pwa');
+    const prevAuditRefs = category.auditRefs;
+    const auditWithWarning = prevAuditRefs[0];
+    auditWithWarning.result = Object.assign({}, category.auditRefs[0].result,
+      {warnings: ['Some warning']});
+    category.auditRefs = [auditWithWarning];
+
+    const auditDOM = renderer.render(category);
+    const passedAuditGroupEl = auditDOM.querySelector('details.lh-passed-audits');
+    const isExpanded = passedAuditGroupEl.hasAttribute('open');
+    assert.ok(isExpanded, 'Passed audit group with warning should be expanded by default');
+
+    category.auditRefs = prevAuditRefs;
```

Comment:
What is this line doing?  Cleaning up auditRefs?  Can this make a deepCopy of category instead so it doesn't modify it for each test (like `'doesnt create a passed section if there were 0 passed'`)?  e.g.
```javascript
const origCategory = sampleResults.reportCategories.find(c => c.id === 'pwa');
const category = JSON.parse(JSON.stringify(origCategory));
```

Reply:
I followed a pattern I saw in the `handles markdown in category descriptions a category` test case. Since `sampleResults` is being modified, and is initialized once and used in all the tests, I needed to restore the state of the `auditRefs` property for the given category.

I see it is also done in the way you suggested here: `doesnt create a passed section if there were 0 passed`. So there's two different methods for avoiding unwanted state change. If one is preferred, I can change the other instance to match.